### PR TITLE
enabling strictNullChecks, strictFunctionTypes, and noImplicitAny

### PR DESCRIPTION
**Pull request by coffeeandwork**

### CODE REVIEW

**Comment by coffeeandwork** [comment]
@leonardorafael any suggestions on PR?

File: src/cdn/beer.ts [hunk 1]
```diff
@@ -65,27 +66,29 @@ export default (() => {
...
     return element?.nextElementSibling;
   }
 
-  function parent (element: Element): Element {
+  function parent (element: Element): Element | null {
     return element?.parentElement;
   }
 
-  function create (json: any): HTMLElement {
+  function create (json: { [key: string]: string | null }): HTMLElement {
```
Comment (leonardorafael) [inline src/cdn/beer.ts:85]: I don't understand this params. There are no type to represent that we need here?

**Comment by coffeeandwork** [comment]
Thank you @leonardorafael . I will review and get back to you shortly.

File: src/cdn/beer.ts [hunk 2]
```diff
@@ -238,7 +245,11 @@ export default (() => {
   }
 
   function open (from?: Element, to?: Element, options?: any, e?: Event): any {
-    if (!to) to = query(from.getAttribute("data-ui"));
+    if (!from) return;
+    const dataUi = from.getAttribute("data-ui");
+    if (!dataUi) return;
+    if (!to) to = query(dataUi) ?? undefined;
+    if (!to) return;
```
[the remark this answers is not in the document]
Comment (coffeeandwork) [reply src/cdn/beer.ts:248]: Does making the function `async` solve the unexpected behavior? If not, perhaps I don't understand the unexpected behavior you are referring to. Please advise.

**Review by coffeeandwork — COMMENTED** [review]
Thank you @leonardorafael. I have address all of your comments. Please take another look.

**Review by myrp-leonardo — APPROVED** [review]

**Review by leonardorafael — APPROVED** [review]
👍

**Comment by coffeeandwork** [comment]
thank you to @myrp-leonardo and @leonardorafael.

**Pull request merged**